js/source/raster_tile_source.js

```diff
@@ -37,15 +40,27 @@ class RasterTileSource extends Evented {
         this.map = map;
     }
 
+    setBounds(bounds) {
+      this.bounds = bounds;
+      if (bounds) {
+        this.tileBounds = new TileBounds(bounds, this.minzoom, this.maxzoom);
+      }
+    }
+
     serialize() {
         return {
             type: 'raster',
             url: this.url,
             tileSize: this.tileSize,
-            tiles: this.tiles
+            tiles: this.tiles,
+            bounds: this.bounds,
```

Comment:
This is why I kept both `this.bounds` as unmodified from what the client provides as well as storing the initialized object as `this.tileBounds`.  The naming at the moment leaves something to be desired for sure.  It seems preferable to me that TileBounds not need to worry about how to round-trip whatever source format the tile source uses?